dagobah/core/core.py

```diff
@@ -690,6 +733,26 @@ def _serialize(self, include_run_logs=False, strict_json=False):
             result = json.loads(json.dumps(result, cls=StrictJSONEncoder))
         return result
 
+
+    def _implements_function(obj, functions):
+        """ Checks for the existence of a method or list of methods """
+        for expected_method in functions:
+            if not (hasattr(obj, expected_method) and
+                    callable(getattr(obj, expected_method))):
+                return False
+        return True
```

Comment:
By convention, the first arg to an instance method should always be named `self`. I think this function is also candidate for not existing in the class at all and instead being a standalone function in a module somewhere.